pkg/object/mqttproxy/mqtt_test.go

```diff
@@ -841,8 +845,18 @@ func (ts *testServer) addHandlerFunc(pattern string, f http.HandlerFunc) {
 	ts.mux.HandleFunc(pattern, f)
 }
 
-func (ts *testServer) start() {
+func (ts *testServer) start() error {
 	go ts.srv.ListenAndServe()
+	// Poll server until it is ready
+	for t := 0; t < 25; t++ {
+		time.Sleep(50 * time.Millisecond)
+		req, _ := http.NewRequest(http.MethodGet, "http://localhost:"+ts.addr, nil)
```

Comment:
This `ts.addr` here is `:8888`, you combine with "http://localhost:" will become "http://localhost::8888", replace it with simple ts.addr will work
```suggestion
		req, _ := http.NewRequest(http.MethodGet, ts.addr, nil)
```